core/operations/management/bucket_create.cxx

```diff
@@ -48,27 +48,35 @@ bucket_create_request::encode_to(encoded_request_type& encoded, http_context& /*
         case couchbase::core::management::cluster::bucket_type::unknown:
             break;
     }
-    encoded.body.append(fmt::format("&ramQuotaMB={}", bucket.ram_quota_mb));
-    if (bucket.bucket_type != couchbase::core::management::cluster::bucket_type::memcached) {
-        encoded.body.append(fmt::format("&replicaNumber={}", bucket.num_replicas));
+    if (bucket.ram_quota_mb == 0) {
+        encoded.body.append(fmt::format("&ramQuotaMB={}", 100)); // If not explicitly set, set to prior default value of 100
```

Comment:
I have asked the team if we possibly want some logic here to set the default to 1024 if the storage backend is set and set to magma.  As is, 100 MB is insufficient for a magma bucket and I am quite sure the server will return an error.

Reply:
My cbdyncluster says minumum is 256, and it seems this setting might be dynamic.

https://github.com/couchbase/ns_server/blob/96072ad51c6739ec75f80737756f4db7bb7af13d/src/menelaus_web_buckets.erl#L1256-L1261

<img width="520" alt="image" src="https://github.com/couchbaselabs/couchbase-cxx-client/assets/7280/a6dca338-ce26-49f8-a91d-73e365a38744">

Reply:
256MB is for serverless

https://github.com/couchbase/ns_server/blob/96072ad51c6739ec75f80737756f4db7bb7af13d/etc/serverless_profile#L11